packages/import/src/helpers/labelManager.ts

```diff
@@ -0,0 +1,283 @@
+import { IssueLabel, LinearClient } from "@linear/sdk";
+import { ImportResult } from "../types";
+import _ from "lodash";
+
+type Id = string;
+const WORKSPACE_ID = "workspace";
+
+class LabelManager {
+  private nameToLabel: Record<string, { [teamId: Id | typeof WORKSPACE_ID]: Label }> = {};
+  private idToLabel: Record<Id, { [teamId: Id | typeof WORKSPACE_ID]: Label }> = {};
+
+  public constructor(
+    existingLabels: IssueLabel[],
+    private teamId: Id
+  ) {
+    this.initializeLabels(existingLabels);
+  }
+
+  public getLabelByName(name: string, teamId: Id | typeof WORKSPACE_ID): Label | undefined {
+    return this.nameToLabel[name.toLowerCase()]?.[teamId] ?? this.nameToLabel[name]?.[WORKSPACE_ID];
+  }
+
+  public getLabelById(id: Id, teamId: Id | typeof WORKSPACE_ID): Label | undefined {
+    return this.idToLabel[id]?.[teamId] ?? this.idToLabel[id]?.[WORKSPACE_ID];
+  }
+
+  private getLabel<T>(props: { name: string } | { id: Id }, checkInstance: (label: Label) => boolean): T | undefined {
+    let label: Label | undefined;
+    if ("name" in props) {
+      label = this.getLabelByName(props.name, this.teamId);
+    } else {
+      label = this.getLabelById(props.id, this.teamId);
+    }
+    return (label && checkInstance(label) ? label : undefined) as T | undefined;
+  }
+
+  public getGroupLabel(props: { name: string } | { id: Id }): LabelGroup | undefined {
+    return this.getLabel<LabelGroup>(props, label => label instanceof LabelGroup);
+  }
+
+  public getRootLabel(props: { name: string } | { id: Id }): Label | undefined {
+    return this.getLabel<Label>(props, label => label instanceof Label);
+  }
```

Comment:
I think we should try to document the public methods and the different terms we use (GroupLabel, RootLabel, SubgroupLabel)